core/src/main/scala/dogs/BiPartition.scala

```diff
@@ -0,0 +1,36 @@
+/**
+  * Created by anicolaspp on 4/10/16.
+  */
+
+package dogs
+
+import Predef._
+import simulacrum.typeclass
+import scala.{inline,Iterable}
+import java.lang.{String,StringBuilder}
+import scala.annotation.{tailrec}
+import dogs.syntax.birds._
+import cats._
+
+
+@typeclass trait BiPartition[A] {
+  def partition(predicate: A => Boolean): (List[A], List[A])
```

Comment:
But I guess in general this typeclass doesn't make sense to me, It says "for some type A, I can take a function A => Boolean, and produce from that two Lists, but it doesn't say where the source of these values is? Why is is parameterized by A?

Is there some value of this typeclass outside of quicksort?